prom.go

```diff
@@ -45,6 +52,61 @@ type Prometheus struct {
 	PathMap     pmap
 }
 
+// IncrementGaugeValue increments a custom gauge
+func (p *Prometheus) IncrementGaugeValue(name string, labelValues []string) error {
+	p.customGauges.Lock()
+	defer p.customGauges.Unlock()
```

Comment:
```suggestion
	p.customGauges.RLock()
	defer p.customGauges.RUnlock()
```

Since we're just reading the content of the map, there's no need to lock for write operations. 
RLock also has the advantage that the lock can be acquired multiple times by the same thread.